EIPS/eip-2477.md

```diff
@@ -0,0 +1,211 @@
+---
+eip: 2477
+title: Non-Fungible Token Metadata Integrity
+author: Kristijan Sedlak (@xpepermint), William Entriken <github.com@phor.net>
+discussions-to: https://github.com/ethereum/EIPs/pull/2477
+type: Standards Track
+category: ERC
+status: Draft
+created: 2020-01-02
+requires: 721, 1155, 165
+---
+
+## Simple Summary
+
+This specification defines a mechanism by which clients may verify that a fetched token metadata document has been delivered without unexpected manipulation.
+
+This is the Web3 counterpart of the W3C Subresource Integrity (SRI) specification.
+
+## Abstract
+
+An interface `ERC2477` with two functions `tokenURIIntegrity` and ``tokenURISchemaIntegrity` are specified for smart contracts and a narrative is provided to explain how this improves the integrity of the token metadata documents.
+
+## Motivation
+
+Tokens are being used in many applications to represent, trace and provide access to assets off-chain. These assets include in-game digital items in mobile apps, luxury watches and products in our global supply chain, among many other creative uses.
+
+Several token standards allow attaching metadata to specific tokens using a URI (RFC 3986) and these are supported by the applications mentioned above. These metadata standards are:
+
+* ERC-721 metadata extension (`ERC721Metadata`)
+* ERC-1155 metadata extension (`ERC1155Metadata_URI`)
+* ERC-1046 (DRAFT) ERC-20 Metadata Extension
+
+Although all these standards allow storing the metadata entirely on-chain (using the "data" URI, RFC 2397), or using a content-addressible system (e.g. IPFS's Content IDentifiers [sic]), every implementation we have found is using Uniform Resource Locators. These URLs provide no guarantees of content correctness or immutability. This standard adds such guarantees.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Smart contracts implementing the ERC-2477 standard MUST implement the `ERC2477` interface.**
+
+```solidity
+pragma solidity ^0.6.0;
+
+/// @title ERC-2477 Non-Fungible Token Standard
+/// @dev See https://eips.ethereum.org/EIPS/eip-2477
+/// @dev The ERC-165 identifier for this interface is 0x#######. //TODO: FIX THIS
+interface ERC2477 /* is ERC165 */ {
+    /**
+     * @notice Get a cryptographic hash for the specificed token metadata
+     * @param tokenId Identifier for a specific token
+     * @return digest Bytes returned from the hash algorithm
+     * @return hashAlgorithm The name of the cryptographic hash algorithm
+     */
+    function tokenURIIntegrity(uint256 tokenId) external view returns(bytes digest, string hashAlgorithm);
+    
+    /**
+     * @notice Get a cryptographic hash for the specificed token metadata's schema
```

Comment:
Spelling: specificed -> specified
Suggest rewriting to:
`* @notice Get the cryptographic hash of the specified tokenID's metadata schema`